scripts/airflow/tasks/build_artery_centreline/build_artery_centreline.sql

```diff
@@ -1,26 +1,42 @@
-drop table if exists prj_volume.artery_centreline;
-create table prj_volume.artery_centreline as (
-  (
-    select
-      pvat.arterycode,
-      pvat.centreline_id,
-      1 as centreline_type,
-      ST_ClosestPoint(gc.geom, ST_Centroid(gc.geom)) as geom
-    from gis.centreline gc
-    join prj_volume.artery_tcl pvat
-    on gc.geo_id = pvat.centreline_id
-    where pvat.artery_type = 1 and pvat.centreline_id is not null
-    order by pvat.arterycode asc
-  ) UNION ALL (
-    select
-      arterycode,
-      cast(int_id as bigint) as centreline_id,
-      2 as centreline_type,
-      ST_SetSrid(ST_MakePoint(longitude, latitude), 4326) as geom
-    from prj_volume.artery_intersections
+CREATE SCHEMA IF NOT EXISTS prj_volume;
+
+CREATE MATERIALIZED VIEW IF NOT EXISTS prj_volume.artery_centreline AS (
+  WITH artery_centreline_raw AS (
+    (
+      select
+        pvat.arterycode,
+        pvat.centreline_id,
+        1 as centreline_type,
+        ST_Transform(ST_ClosestPoint(
+          ST_Transform(gc.geom, 2952),
+          ST_Transform(ST_Centroid(gc.geom), 2952)
+        ), 4326) as geom
+      from gis.centreline gc
+      join prj_volume.artery_tcl pvat
+      on gc.geo_id = pvat.centreline_id
+      where pvat.artery_type = 1 and pvat.centreline_id is not null
+      order by pvat.arterycode asc
+    ) UNION ALL (
+      select
+        arterycode,
+        cast(int_id as bigint) as centreline_id,
+        2 as centreline_type,
+        ST_SetSrid(ST_MakePoint(longitude, latitude), 4326) as geom
+      from prj_volume.artery_intersections
+    )
   )
+  SELECT
+    arterycode,
+    centreline_id,
+    centreline_type,
+    MODE() WITHIN GROUP(ORDER BY geom) AS geom
+  FROM artery_centreline_raw
+  GROUP BY arterycode, centreline_id, centreline_type
 );
-create index artery_centreline_arterycode on prj_volume.artery_centreline (arterycode);
-create index artery_centreline_centreline_id on prj_volume.artery_centreline (centreline_id);
-create index artery_centreline_geom on prj_volume.artery_centreline using gist (geom);
-create index artery_centreline_srid3857_geom ON prj_volume.artery_centreline using gist (ST_Transform(geom, 3857));
+create unique index IF NOT EXISTS artery_centreline_arterycode on prj_volume.artery_centreline (arterycode);
+create index IF NOT EXISTS artery_centreline_centreline on prj_volume.artery_centreline (centreline_type, centreline_id);
+create index IF NOT EXISTS artery_centreline_geom on prj_volume.artery_centreline using gist (geom);
+create index IF NOT EXISTS artery_centreline_srid3857_geom ON prj_volume.artery_centreline using gist (ST_Transform(geom, 3857));
+create index IF NOT EXISTS artery_centreline_srid2952_geom ON prj_volume.artery_centreline using gist (ST_Transform(geom, 2952));
```

Comment:
note that `PoiDAO` still uses the old SRID 26917 projection - please change in follow-up work.